### PR TITLE
Fix all the issues that I need for pcb

### LINKED ISSUE #40: `fn func(mut thing: T)` gets turned into `void func(T mut thing)`
instead of `void func(T thing)`.

### PR DESCRIPTION
**Pull request by ubsan**
Fixes #40, #39, and #32.

### CODE REVIEW

**Comment by ubsan** [comment]
(https://github.com/ubsan/pcb, if you were wondering)

File: src/parse.rs [hunk 1]
```diff
@@ -305,7 +305,13 @@ fn parse_fn(item: &ast::Item) -> Result<Option<String>, Error> {
         let has_args = !fn_decl.inputs.is_empty();
 
         for arg in &fn_decl.inputs {
-            let arg_name = print::pprust::pat_to_string(&*arg.pat);
+            use syntax::ast::{PatIdent, BindingMode};
+            let arg_name = match arg.pat.node {
+                PatIdent(BindingMode::ByValue(_), ref ident, None) => {
+                    ident.node.name.to_string()
+                }
+                _ => panic!("hi")
```
Comment (Sean1708) [inline src/parse.rs]: This is an unreachable arm isn't it? If so could you change that to something along the lines of

```rust
_ => unreachable!("cheddar should have already check that the arguments are  by-value patterns"),
```

please.

Comment (ubsan) [inline src/parse.rs]: I did not realize; this was test code that I never actually tested. Sorry :/

**Comment by Sean1708** [comment]
Looks good, thanks @ubsan!

**Pull request merged**